cmd/resolutioncli/fbcsource.go

```diff
@@ -0,0 +1,156 @@
+}
+
+func (es *fbcEntitySource) Filter(ctx context.Context, filter input.Predicate) (input.EntityList, error) {
+	resultSet := input.EntityList{}
+	for _, entity := range es.entities {
+		if filter(&entity) {
+			resultSet = append(resultSet, entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es *fbcEntitySource) GroupBy(ctx context.Context, fn input.GroupByFunction) (input.EntityListMap, error) {
+	resultSet := input.EntityListMap{}
+	for _, entity := range es.entities {
+		keys := fn(&entity)
+		for _, key := range keys {
+			resultSet[key] = append(resultSet[key], entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es *fbcEntitySource) Iterate(ctx context.Context, fn input.IteratorFunction) error {
+	for _, entity := range es.entities {
+		if err := fn(&entity); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getEntities(bundleMetadatas []corev1beta1.BundleMetadata, packageMetadatas []corev1beta1.Package) (input.EntityList, error) {
+	nameToPkg := make(map[string]*corev1beta1.Package, len(packageMetadatas))
+	for i := range packageMetadatas {
+		nameToPkg[packageMetadatas[i].GetName()] = &packageMetadatas[i]
+	}
+
+	entities := input.EntityList{}
+	for _, bundle := range bundleMetadatas {
+		props := map[string]string{}
+
+		for _, prop := range bundle.Spec.Properties {
+			switch prop.Type {
+			case property.TypePackage:
+				// this is already a json marshalled object, so it doesn't need to be marshalled
+				// like the other ones
+				props[property.TypePackage] = string(prop.Value)
+			}
+		}
+
+		imgValue, err := json.Marshal(bundle.Spec.Image)
+		if err != nil {
+			return nil, err
+		}
+		props["olm.bundle.path"] = string(imgValue)
+		catalogScopedPkgName := fmt.Sprintf("%s-%s", bundle.Spec.Catalog.Name, bundle.Spec.Package)
+		bundlePkg := nameToPkg[catalogScopedPkgName]
+		for _, ch := range bundlePkg.Spec.Channels {
+			for _, b := range ch.Entries {
+				catalogScopedEntryName := fmt.Sprintf("%s-%s", bundle.Spec.Catalog.Name, b.Name)
+				if catalogScopedEntryName == bundle.Name {
+					channelValue, _ := json.Marshal(property.Channel{ChannelName: ch.Name, Priority: 0})
+					props[property.TypeChannel] = string(channelValue)
+					entity := input.Entity{
+						ID:         deppy.IdentifierFromString(fmt.Sprintf("%s%s%s", bundle.Name, bundle.Spec.Package, ch.Name)),
+						Properties: props,
+					}
+					entities = append(entities, entity)
+				}
+			}
+		}
+	}
+	return entities, nil
+}
+
+func fetchMetadata(fbcRoot fs.FS) ([]corev1beta1.BundleMetadata, []corev1beta1.Package, error) {
+	fbc, err := declcfg.LoadFS(fbcRoot)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error loading FBC from filesystem: %w", err)
+	}
+
+	// TODO: get rid of this
+	catalog := &corev1beta1.Catalog{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "offline-fbc-catalog",
+		},
+	}
+
+	bundleMetadatas := ToBundleMetadata(fbc, catalog)
+	packages, err := ToPackages(fbc, catalog)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
Should probably use `github.com/operator-framework/operator-registry/alpha/model` instead